src/sources/internal_metrics.rs

```diff
@@ -59,33 +70,57 @@ pub struct TagsConfig {
     pub pid_key: Option<String>,
 }
 
+impl Default for TagsConfig {
+    fn default() -> Self {
+        Self {
+            host_key: default_host_key(),
+            pid_key: None,
+        }
+    }
+}
+
+fn default_scrape_interval() -> Duration {
+    Duration::from_secs_f64(1.0)
+}
+
+fn default_namespace() -> String {
+    "vector".to_owned()
+}
+
+fn default_host_key() -> String {
+    log_schema().host_key().to_owned()
+}
+
 impl_generate_config_from_default!(InternalMetricsConfig);
 
 #[async_trait::async_trait]
 impl SourceConfig for InternalMetricsConfig {
     async fn build(&self, cx: SourceContext) -> crate::Result<super::Source> {
-        if self.scrape_interval_secs == 0.0 {
+        if self.scrape_interval_secs.is_zero() {
             warn!(
                 "Interval set to 0 secs, this could result in high CPU utilization. It is suggested to use interval >= 1 secs.",
             );
         }
-        let interval = time::Duration::from_secs_f64(self.scrape_interval_secs);
-        let namespace = self.namespace.clone();
+        let interval = self.scrape_interval_secs;
 
-        let host_key = match self
-            .tags
-            .host_key
-            .as_deref()
-            .unwrap_or_else(|| log_schema().host_key())
-        {
+        // namespace for created metrics is already "vector" by default.
```

Comment:
Feels like it'd be cleaner to update the `InternalMetrics` struct to take this as a `String`, rather than an `Option<String>`?

Reply:
Yep, not a required change - just struck me as cleaner and easier to follow. If it turns out to not be nicer, happy to let it through as-is